storage/libradb/src/transaction_store/mod.rs

```diff
@@ -19,6 +25,25 @@ impl TransactionStore {
         Self { db }
     }
 
+    /// Gets the version of a transaction by the sender `address` and `sequence_number`.
+    pub fn lookup_transaction_by_account(
+        &self,
+        address: AccountAddress,
+        sequence_number: u64,
+        ledger_version: Version,
+    ) -> Result<Option<Version>> {
+        if let Some(version) = self
+            .db
+            .get::<TransactionByAccountSchema>(&(address, sequence_number))?
+        {
+            if version <= ledger_version {
+                return Ok(Some(version));
+            }
```

Comment:
If version > ledger_version, we have a big problem - should you log something about this?

Reply:
It can be "when the read API call is running (ledger_version determined), a new txn for the account is commited". -- I think it's not a problem.